packages/@lwc/module-resolver/src/index.ts

```diff
@@ -4,117 +4,147 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-/* eslint-env node */
 
 import path from 'path';
 import fs from 'fs';
 import {
+    RegistryEntry,
+    AliasModuleRecord,
+    InnerResolverOptions,
+    ModuleRecord,
+    DirModuleRecord,
+    ModuleResolverConfig,
+    NpmModuleRecord,
+} from './types';
+import {
+    createRegistryEntry,
+    findFirstUpwardConfigPath,
+    validateModuleRecord,
+    isAliasModuleRecord,
+    isDirModuleRecord,
+    isNpmModuleRecord,
+    validateImportee,
+    getLwcConfig,
     getModuleEntry,
     normalizeConfig,
-    loadConfig,
-    isString,
+    validateNpmConfig,
     mergeModules,
-    LWC_CONFIG_FILE,
 } from './utils';
 
-export interface RegistryEntry {
-    entry: string;
-    specifier: string;
-}
+function resolveModuleFromAlias(
+    moduleRecord: AliasModuleRecord,
+    opts: InnerResolverOptions
+): RegistryEntry {
+    const { name: specifier, path: modulePath } = moduleRecord;
+    const entry = path.resolve(opts.rootDir, modulePath);
+    if (!fs.existsSync(entry)) {
+        throw new Error(
+            `Unable to find AliasModuleRecord for "${specifier}". File ${entry} does not exist`
+        );
+    }
 
-export interface ModuleRegistryMap {
-    [key: string]: RegistryEntry;
-}
-export interface ModuleRecordObject {
-    name: string;
-    path: string;
+    return createRegistryEntry(entry, specifier, opts);
 }
 
-export interface ModuleResolverConfig {
-    rootDir: string;
-    modules: ModuleRecord[];
-}
+function resolveModuleFromDir(
+    specifier: string,
+    moduleRecord: DirModuleRecord,
+    opts: InnerResolverOptions
+): RegistryEntry | undefined {
+    const { dir } = moduleRecord;
+    const absModuleDir = path.isAbsolute(dir) ? dir : path.join(opts.rootDir, dir);
+    const [ns, name] = specifier.split('/');
```

Comment:
What happens if the specifier is `foo/bar/baz` the `baz` section will be ignored and we will attempt to resolve `foo/bar` which is inaccurate.